components/DurationComponent.tsx

```diff
@@ -38,15 +38,15 @@ export function DurationComponent(props: {
         }}
       />
       <div style={{ padding: 8 }} />
-      <SingleSelect
+      <Autocomplete
         disabled={props.disabled}
-        items={getTimeDefinitionValues()}
-        handleSelectedItemChange={(i) => {
-          const apiValue = getTimeDefinitionValue(i.selectedItem);
+        options={getTimeDefinitionValues()}
+        onInputChange={(i) => {
+          const apiValue = getTimeDefinitionValue(i);
           setUnit(apiValue);
           props.onChangeTime({ time: time, unit: apiValue });
         }}
-        value={getTimeDefinitionDisplayName(unit)}
+        selectedOptions={[getTimeDefinitionDisplayName(unit)]}
         label="Unit"
       />
```

Comment:
This has some TypeScript errors when setting `strict: true` in `tsconfig.json`.

Reply:
These errors was always an issue from the typing of the callback prop, and this callback is passed down multiple files. This should be fixed in #668